vpr/src/noc/noc_traffic_flows.h

```diff
@@ -0,0 +1,307 @@
+     * retrieved traffic flow cannot be modified but can be used to
+     * retireve information such as the routers involved.
+     * 
+     * @param traffic_flow_id The unique identifier (NocTrafficFlowId)
+     * of the traffic flow to retrieve.
+     * @return const t_noc_traffic_flow& The traffic flow represented by
+     * the provided identifier.
+     */
+    const t_noc_traffic_flow& get_single_noc_traffic_flow(NocTrafficFlowId traffic_flow_id) const;
+
+    /**
+     * @brief Get a vector of all traffic flows that have a given router
+     * block in the clustered netlist as the source (starting point) in the 
+     * flow.
+     * 
+     * @param source_router_id A unique identifier that represents the
+     * a router block in the clustered netlist. This router block will
+     * be the source router in the retrieved traffic flow.
+     * @return const std::vector<NocTrafficFlowId>* A vector of traffic 
+     * flows that have the input router block parameter as the source in the
+     * flow.
+     */
+    const std::vector<NocTrafficFlowId>* get_traffic_flows_associated_to_source_router(ClusterBlockId source_router_id) const;
+
+    /**
+     * @brief Get a vector of all traffic flows that have a given router
+     * block in the clustered netlist as the sink (end point) in the 
+     * flow.
+     * 
+     * @param sink_router_id A unique identifier that represents the
+     * a router block in the clustered netlist. This router block will
+     * be the sink router in the retrieved traffic flow.
+     * @return const std::vector<NocTrafficFlowId>* A vector of traffic 
+     * flows that have the input router block parameter as the sink in the
+     * flow.
+     */
+    const std::vector<NocTrafficFlowId>* get_traffic_flows_associated_to_sink_router(ClusterBlockId sink_router_id) const;
+
+    /**
+     * @brief Given a traffic flow, determine whether it has been processed
+     * or not. By processed, it checks to see whether the traffic flow has
+     * been routed. This function should be used before routing a traffic
+     * flow.
+     * 
+     * @param traffic_flow_id A unique identifier that represents a traffic
+     * flow
+     * @return true Represents the case where the traffic flow has been
+     * processed
+     * @return false Represents the case where the traffic flow has not
+     * been processed 
+     */
+    bool get_traffic_flow_processed_status(NocTrafficFlowId traffic_flow_id);
+
+    /**
+     * @brief Gets the number of unique router blocks in the
+     * clustered netlist that were used within the user provided
+     * traffic flows description. 
+     * 
+     * @return int The total number of unique routers used in
+     * the traffic flows provided by the user.
+     */
+    int get_number_of_routers_used_in_traffic_flows(void);
+
+    // setters
+
+    /**
+     * @brief Given a set of parameters that describe a traffic
+     * flow, create it an add it to the vector of traffic flows in the
```

Comment:
May want to be a little shorter in comment, so they don't get out of date as fast. E.g.: add the specified traffic flows to the vector of flows, and update the router->flow look up tables.

Reply:
Done